### PR TITLE
(dynamic alert thresholds) Force recompilation of Stumpy library during bootup

### PR DESCRIPTION
**Pull request by ram-senth**
Use mode preloading mechanism to force numba compilation for Stumpy library

### CODE REVIEW

File: src/seer/anomaly_detection/anomaly_detection.py [hunk 1]
```diff
@@ -27,6 +33,29 @@
...
+    @sentry_sdk.trace
+    def __init__(self):
+        """
+        Force Stumpy compilation by making a dummy call to the library.
+
+        Note: compilation triggered here is very specific to the exact values of parameters ignore_trivial, normalize etc. A
+            future call with different values for one or more parameter will still trigger a recompilation.
+        """
+        data = np.arange(10.0)
+        try:
```
Comment (trillville) [inline src/seer/anomaly_detection/anomaly_detection.py:45]: does this windowsizeselector logic need to be here?

**Review by trillville — APPROVED** [review]
lgtm - how long does this take? do we need to change anything on the ops side WRT bootup logic to accommodate?

File: src/seer/anomaly_detection/anomaly_detection.py (hunk 1 above)
Comment (ram-senth) [reply src/seer/anomaly_detection/anomaly_detection.py:45]: That is a good point, it doesn't have to be. I was experimenting and forgot to remove that.

**Comment by ram-senth** [comment]
> lgtm - how long does this take? do we need to change anything on the ops side WRT bootup logic to accommodate?

Interestingly, it doesn't take much time actually, less than 100 ms. Seems like it is a function of the size of the time series used in the first call.

**Pull request merged**